### ISSUE TITLE
Regex $ (end of string) should not match \n (end of line)

### ISSUE THREAD

**Issue by zendu** [dotnet/corefx#36467]
From [MSDN](https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-options#default-options)

> The `^` and `$` language elements match the beginning and end of the input string.

I am using pattern (in code below) to ensure that input is valid email. It matches inputs like `test@test.com` and rejects inputs that are not valid email. Interestingly, some times input contains a trailing new line character. I would expect the pattern to reject inputs like `test@test.com\n`. However, it seems that regex match `$` with `\n` and returns true. Validation succeeds in my system but the email address is rejected by other systems. This is expected as input with trailing new line is clearly not a valid email address.

**Why does `Regex.IsMatch()` returns `True` when it should not?**

### Sample program

```c#
    using System;
    using System.Text.RegularExpressions;
					
    public class Program
    {
        public static void Main()
        {
            var input = "test@test.com\n";
            var pattern = @"^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5}){1,25}$";
            
            var test1 = Regex.IsMatch(input, pattern, RegexOptions.ExplicitCapture);
            var test2 = Regex.IsMatch(input, pattern, RegexOptions.ExplicitCapture | RegexOptions.Singleline);
            var test3 = Regex.IsMatch(input, pattern, RegexOptions.ExplicitCapture | RegexOptions.Multiline);
            var test4 = Regex.IsMatch(input, pattern);
            
            Console.WriteLine(test1);
            Console.WriteLine(test2);
            Console.WriteLine(test3);
            Console.WriteLine(test4);
        }
    }
```

### Expected output
```
    False
    False
    True
    False
```
### Actual output
```
    True
    True
    True
    True
```

**Comment by kislovs** [comment]
@zendu It is correct behavior. Check this out: [Anchors in Regular Expressions, MSDN](https://docs.microsoft.com/en-us/dotnet/standard/base-types/anchors-in-regular-expressions#end-of-string-or-line-).

**Comment by zendu** [comment]
Thanks @kislovs. I modifier pattern to `@"^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5}){1,25}\z"` to get desired result i.e. not to match when input ends in new line character.

**Comment by kislovs** [comment]
@karelz why did you add this to the 3.0 milestone?

**Comment by karelz** [comment]
Because it was closed during 3.0 milestone. Would you expect different milestone?

**Comment by kislovs** [comment]
@karelz, got it. I thought you add milestones when you want to make some changes to issue topic.
